src/galaxy/GalaxyConfig.cpp

```diff
@@ -6,13 +6,14 @@
 
 GalaxyConfig::GalaxyConfig()
 {
-	// set defaults
-	std::map<std::string, std::string> &map = m_map[""];
-	map["GalaxyExploredMax"] = "90";
-	map["GalaxyExploredMin"] = "65";
-	map["GalaxyExploredMix"] = "40";
-
 	Read(FileSystem::userFiles, "galaxy.ini");
 
-	Save();
+	// set defaults
+	SetInt("GalaxyExploredMax", 90);
+	SetInt("GalaxyExploredMin", 65);
+	SetInt("GalaxyExploredMix", 40);
```

Comment:
Since this is after the `Read()` call, these are no longer defaults but overrides, preventing the user from tweaking galaxy generation settings. Please restore the prior direct assignment as it is significantly more efficient (removes 2x unneeded lookups for the `""` section) and instead directly set `m_unsaved` if no `galaxy.ini` file exists.

Feel free to set the unsaved flag inside of `Read()` if the specified configuration file cannot be read for any reason.

Reply:
This code is called during setup, so efficiency is largely immaterial here. Furthermore this all happens in memory, so I challenge you to find any significantly measurable performance impact...

----

That said, you're right, these are now overrides instead of defaults. I'll apply your suggested fix and test it. Thank you for pointing my mistake out.

Reply:
Reworked in af5d2f37af3136fd9387596c1407c6dd161eec1c